### PR TITLE
`fiberplane-charts`: Handle gaps in metrics

### PR DESCRIPTION
**Pull request by arendjr**
# Description

Detects and handles gaps in line charts.

Fixes: https://github.com/autometrics-dev/explorer/issues/35

# Checklist

- [x] The changes have been tested to be backwards compatible.
- ~~[ ] The OpenAPI schema and generated client have been updated.~~
- ~~[ ] New models module has been added to api generator xtask~~
- [x] New types/fields are [well-documented and annotated](/CONTRIBUTING.md#adding-types-and-their-annotations).
- ~~[ ] The CHANGELOG is updated.~~

> Please link any related PRs in other repositories that depend on this:

After merging, please merge related PRs ASAP, so others don't get blocked.

### CODE REVIEW

File: fiberplane-charts/src/CoreChart/ChartShape/PointShape.tsx [hunk 1]
```diff
@@ -15,9 +15,9 @@ export const PointShape = memo(function PointShape<P>({
 }: Props<P>): JSX.Element {
   return (
     <circle
-      x={scales.xScale(point.x)}
-      y={scales.yScale(point.y)}
-      radius={focused ? 2 : 1}
+      cx={scales.xScale(point.x)}
```
Comment (brettimus) [inline fiberplane-charts/src/CoreChart/ChartShape/PointShape.tsx:18]: I didn't catch this in the last review! Just added a fix here to render points properly

File: fiberplane-charts/src/Mondrian/generator/generateLineChartFromTimeseries.ts [hunk 2]
```diff
@@ -54,18 +59,31 @@ function getShapes(
...
+              ...getPointForMetric(metric, xAxis, yAxis),
+            },
+          ];
+    }
+    default: {
+      const lines = splitIntoContinuousLines(metrics, interval ?? undefined);
+      return lines.map((line) =>
+        // If the line only containes one metric value, render it as a point
+        // Otherwise, render a line
+        line.length === 1
```
Comment (brettimus) [inline fiberplane-charts/src/Mondrian/generator/generateLineChartFromTimeseries.ts:78]: I also added this logic as a self-test 😁 

We treat any discontiguous line segment as a point if it only contains a single value

Comment (brettimus) [reply fiberplane-charts/src/Mondrian/generator/generateLineChartFromTimeseries.ts:78]: (effectively scope creeping https://github.com/autometrics-dev/explorer/issues/36 into this PR 🙈 )

Comment (arendjr) [reply fiberplane-charts/src/Mondrian/generator/generateLineChartFromTimeseries.ts:78]: Perfect, makes sense 👍

**Review by brettimus — APPROVED** [review]

**Pull request merged**